src/main/java/org/opensearch/flowframework/transport/ReprovisionWorkflowTransportAction.java

```diff
@@ -270,27 +270,45 @@ private void executeWorkflowAsync(
         ActionListener<WorkflowResponse> listener
     ) {
         try {
-            threadPool.executor(PROVISION_WORKFLOW_THREAD_POOL).execute(() -> { executeWorkflow(template, workflowSequence, workflowId); });
+            threadPool.executor(PROVISION_WORKFLOW_THREAD_POOL).execute(() -> {
+                updateTemplate(template, workflowId);
+                executeWorkflow(workflowSequence, workflowId);
+            });
         } catch (Exception exception) {
             listener.onFailure(new FlowFrameworkException("Failed to execute workflow " + workflowId, ExceptionsHelper.status(exception)));
         }
     }
 
     /**
-     * Executes the given workflow sequence
+     * Replace template document
      * @param template The template to store after reprovisioning completes successfully
+     * @param workflowId The workflowId associated with the workflow that is executing
+     */
+    private void updateTemplate(Template template, String workflowId) {
+        flowFrameworkIndicesHandler.updateTemplateInGlobalContext(workflowId, template, ActionListener.wrap(templateResponse -> {
+            logger.info("Updated template for {}", workflowId, State.COMPLETED);
+        }, exception -> {
+            String errorMessage = "Failed to update use case template for " + workflowId;
+            logger.error(errorMessage, exception);
+        }),
+            true  // ignores NOT_STARTED state if request is to reprovision
+        );
+    }
+
+    /**
+     * Executes the given workflow sequence
      * @param workflowSequence The topologically sorted workflow to execute
      * @param workflowId The workflowId associated with the workflow that is executing
      */
-    private void executeWorkflow(Template template, List<ProcessNode> workflowSequence, String workflowId) {
+    private void executeWorkflow(List<ProcessNode> workflowSequence, String workflowId) {
         String currentStepId = "";
         try {
             Map<String, PlainActionFuture<?>> workflowFutureMap = new LinkedHashMap<>();
             for (ProcessNode processNode : workflowSequence) {
                 List<ProcessNode> predecessors = processNode.predecessors();
                 logger.info(
                     "Queueing process [{}].{}",
-                    processNode.id(),
+                    processNode.id() + processNode.workflowStep().getName(),
```

Comment:
Have some sort of delimiter here, otherwise these two strings will run together.  As written assuming we're using the semantic search template, this will appear as `Queueing process [create_ingest_pipelineupdate_ingest_pipeline].`.  

Not sure the best way to do this but perhaps `[create_ingest_pipeline (update_ingest_pipeline)]`?  Open to other ideas.